### PR TITLE
ci: Add semantic release action flow

### LINKED ISSUE #575: Implement semantic release
### Problem

Currently releases are created manually

### Solution

Implement semantic releases

### Alternatives

-

### PR DESCRIPTION
**Pull request by yaroslav-007**
**Description**:
Adding GIthub action workflow with semantic release. Some specifics:
 - Breaking Changes are set to increment the minor version
 - The Github action workflow is not automatically triggered (you should run it manually)

**Related issue(s)**:

Fixes #575

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by georgi-l95 — DISMISSED** [review]
LG

File: .github/workflows/flow-semantic-release.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+on:
+  workflow_dispatch:
+
+jobs:
+  release:
+    permissions:
+      contents: write
+      issues: write
+      pull-requests: write
+    runs-on: ubuntu-latest
```
Comment (rbarkerSL) [inline .github/workflows/flow-semantic-release.yml:12]: This should be using the self-hosted runner:

`runs-on: [self-hosted, Linux, medium, ephemeral]`

File: .github/workflows/flow-semantic-release.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 20
+      - run: npm ci
+      - run: npm run build
+      - run: npx semantic-release
```
Comment (rbarkerSL) [inline .github/workflows/flow-semantic-release.yml:20]: Couple things here:
- Every step needs a name field
- Anywhere you're using an action you should be using a pinned commit.

```Yaml
# Current:
      - uses: actions/checkout@v2
      - uses: actions/setup-node@v2
        with:
           node-version: 20
      - run: npm ci
      - run: npm run build
      - run: npx semantic-release

# Becomes:
      - name: Checkout Code
        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
      - name: Configure Node JS
        uses: actions/setup-node@60edb5dd545a775178f52524783378180af0d1f8 # v4.0.2
        with:
           node-version: 20
      - name: Clean Install
        run: npm ci
      - name: Build Source
        run: npm run build
      - name: Run semantic-release
        run: npx semantic-release
```

**Review by rbarkerSL — CHANGES_REQUESTED** [review]

File: .github/workflows/flow-semantic-release.yml (hunk 2 above)
Comment (yaroslav-007) [reply .github/workflows/flow-semantic-release.yml]: Hello @rbarkerSL,
Thanks for taking the time to review this PR. I have addressed you comment accordingly.

File: .github/workflows/flow-semantic-release.yml (hunk 1 above)
Comment (yaroslav-007) [reply .github/workflows/flow-semantic-release.yml]: Hello @rbarkerSL,
Thanks for taking the time to review this PR. I have addressed your comment.

**Review by rbarkerSL — APPROVED** [review]
LGTM

**Review by georgi-l95 — DISMISSED** [review]

**Review by isavov — APPROVED** [review]

**Review by georgi-l95 — APPROVED** [review]

**Review by isavov — APPROVED** [review]

**Pull request merged**